Some Makefile modifications to build our Debian package. We also allow the use
of CFLAGS.
==========================================================================
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,11 @@
 TGT=noip2
-CC=gcc
+CC=$(DEB_HOST_GNU_TYPE)-gcc
+CFLAGS = -Wall -g
+ifneq (,$(filter noopt,$(DEB_BUILD_OPTIONS)))
+	CFLAGS += -O0
+else
+	CFLAGS += -O2
+endif
 PKG=noip-2.1.tgz
 
 PREFIX=/usr/local
@@ -8,7 +14,7 @@
 
 # these defines are for Linux
 LIBS=
-ARCH=linux
+ARCH=$(DEB_HOST_ARCH_OS)
 
 # for Mac OS X and BSD systems that have getifaddr(), uncomment the next line
 #ARCH=bsd_with_getifaddrs
@@ -22,7 +28,7 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} ${CFLAGS} -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
 
 install: ${TGT} 
 	if [ ! -d ${BINDIR} ]; then mkdir -p ${BINDIR};fi
